### PR TITLE
JRU_04_01

### PR DESCRIPTION
**Pull request by Maksym82564**

### CODE REVIEW

File: src/main/java/com/javarush/Main.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        main.sessionFactory.getCurrentSession().close();
+
+        List<Integer> ids = List.of(3, 240, 123, 4, 189, 89, 150, 118, 10, 220);
+
+        long startRedis = System.currentTimeMillis();
+        main.testRedisData(ids);
+        long stopRedis = System.currentTimeMillis();
+
+        long startMysql = System.currentTimeMillis();
+        main.testMysqlData(ids);
```
Comment (Volodymyr95) [inline src/main/java/com/javarush/Main.java]: is it test?

File: src/main/java/com/javarush/Main.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        long redisTime = stopRedis - startRedis;
+        long mysqlTime = stopMysql - startMysql;
+
+        logger.info("Redis time: {} ms", redisTime);
+        logger.info("Mysql time: {} ms", mysqlTime);
+
+        main.shutdown();
+    }
+
+    public Main() {
```
Comment (Volodymyr95) [inline src/main/java/com/javarush/Main.java]: in Main method, u just run app, without any logic

File: src/main/java/com/javarush/Main.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+    private void pushToRedis(List<CityCountry> data) {
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            RedisStringCommands<String, String> sync = connection.sync();
+            for (CityCountry cityCountry : data) {
+                try {
+                    sync.set(String.valueOf(cityCountry.getId()), mapper.writeValueAsString(cityCountry));
+                } catch (JsonProcessingException e) {
+                    logger.error("Couldn't push CityCountry to Redis");
```
Comment (Volodymyr95) [inline src/main/java/com/javarush/Main.java]: add e too logger

File: src/main/java/com/javarush/Main.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        countryDAO = new CountryDAO(sessionFactory);
+
+        mapper = new ObjectMapper();
+    }
+
+    private void pushToRedis(List<CityCountry> data) {
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            RedisStringCommands<String, String> sync = connection.sync();
+            for (CityCountry cityCountry : data) {
+                try {
```
Comment (Volodymyr95) [inline src/main/java/com/javarush/Main.java]: why try inside try?

File: src/main/java/com/javarush/Main.java (hunk 1 above)
Comment (Maksym82564) [reply src/main/java/com/javarush/Main.java]: renamed testMysqlData method

File: src/main/java/com/javarush/Main.java (hunk 2 above)
Comment (Maksym82564) [reply src/main/java/com/javarush/Main.java]: fixed

File: src/main/java/com/javarush/Main.java (hunk 3 above)
Comment (Maksym82564) [reply src/main/java/com/javarush/Main.java]: fixed

File: src/main/java/com/javarush/Main.java (hunk 4 above)
Comment (Maksym82564) [reply src/main/java/com/javarush/Main.java]: removed try inside try

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**